### ISSUE TITLE
shift-left arrows doesn't grow selection to the left

### ISSUE THREAD

**Issue by chmike** [marktext/marktext#513]
### Description

The issue is with selection change when using shift-left or shift-right arrows. 
By default, using shift-left or shift-right arrows changes the right end of the selection. If there is no selection, and the **shift-right** arrow key is pressed, the character at the right is selected and the selection length is updated relative to the shift-left or shift-right arrow key pressed. 

This works as expected and as on Ubuntu text fields.

[Description of the bug or feature]

### Steps to reproduce

The problem is when there is no (more) selected text. A shift-left arrow key press should set the keyboard focus on the left end of the selection. This is not the case. Shift-left and shift-rigth arrows should move the left end of the selection. 
 
It currently seam as if a shift-left arrow key press moves the caret one step to the left and set the selection end to where the caret was. The result is that one character is selected. Since the focus is still on the right end of the selection, typinq shift-left arrow again, removes the selection since its length is reduced to 0. 

**Expected behavior:** [What you expected to happen]

When there is no selection, shift-left arrow must set the keyboard focus on the left end of the selection. The shift-left and shift-right arrows should then shrink and grow the selection by moving the left side. The rationale is that shift arrows must not move the caret. 

In summary, the shift arrows may update the left or right end of the selection. By default it updates the right end. It must update the left end, if **shift-left arrow** is typed and there is no selection (or empty). 

### Versions

- Mark Text: 0.12.25
- OS: Ubuntu 18.04 Gnome

**Comment by fxha** [comment]
Thanks @chmike, I think this is a duplicate of #240. Can you confirm this? If so you can subscribe there for future updates.

**Comment by chmike** [comment]
Yes. It’s the same problem, but not well described.

**Comment by fxha** [comment]
@stefnotch Yes, I cannot reproduce it with the latest version too.  I'm closing this issue now because it's a duplicate of yours.

**Comment by chmike** [comment]
What is the latest version you have tested ? The one I use is 0.12.25. In which one is the problem removed ?

**Comment by fxha** [comment]
@chmike `v0.13.50` is the latest but I don't have it released as a flatpak yet because we have two bug fixes and waiting for a upstream electron fix to release a new version. If you are using Linux, you can download the latest AppImage.

**Comment by chmike** [comment]
I tested it and confirm. Feature #513 is fixed. Great and impressivei job.
How is Marktext related to Typora ? Both editors are very similar. I find this way to edit Markdown text excellent.

**Comment by Jocs** [comment]
@chmike Mark Text was originally inspired by Typora, thinking that this is the new way to edit Markdown. Later, think about why not making an open source editor? Let more people come in and contribute to it, so there is Mark Text.

After that, Mark Text will connect with Google Drive or Drop box, Google email, and more.
